src/CabanaPD_HeatTransfer.hpp

```diff
@@ -0,0 +1,127 @@
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef HEATTRANSFER_H
+#define HEATTRANSFER_H
+
+#include <CabanaPD_Timer.hpp>
+
+namespace CabanaPD
+{
+
+// Peridynamic heat transfer with forward-Euler time integration.
+// Inherits only because this is a similar neighbor-based kernel.
+template <class ExecutionSpace, class ModelType>
+class HeatTransfer : public Force<ExecutionSpace, BaseForceModel>
+{
+  protected:
+    using base_type = Force<ExecutionSpace, BaseForceModel>;
+    using base_type::_half_neigh;
+    using base_type::_timer;
+    using model_type = ModelType;
+    static_assert(
+        std::is_same_v<typename model_type::fracture_type, Elastic> );
+
+    Timer _euler_timer = base_type::_energy_timer;
+    ModelType _model;
+
+  public:
+    HeatTransfer( const bool half_neigh, const ModelType model )
+        : base_type( half_neigh )
+        , _model( model )
+    {
+    }
+
+    template <class TemperatureType, class PosType, class ParticleType,
+              class NeighListType, class ParallelType>
+    void
+    computeHeatTransferFull( TemperatureType& conduction, const PosType& x,
+                             const PosType& u, const ParticleType& particles,
+                             const NeighListType& neigh_list, const int n_local,
+                             ParallelType& neigh_op_tag )
+    {
+        _timer.start();
+
+        auto model = _model;
+        const auto vol = particles.sliceVolume();
+        const auto temp = particles.sliceTemperature();
+
+        auto temp_func = KOKKOS_LAMBDA( const int i, const int j )
+        {
+            double xi, r, s;
+            getDistance( x, u, i, j, xi, r, s );
+
+            const double coeff = model.microconductivity_function( xi );
+            conduction( i ) +=
+                coeff * ( temp( j ) - temp( i ) ) / xi / xi * vol( j );
+        };
+
+        Kokkos::RangePolicy<ExecutionSpace> policy( 0, n_local );
+        Cabana::neighbor_parallel_for(
+            policy, temp_func, neigh_list, Cabana::FirstNeighborsTag(),
+            neigh_op_tag, "CabanaPD::HeatTransfer::computeFull" );
+
+        _timer.stop();
+    }
+
+    template <class ParticleType>
+    void forwardEuler( const ParticleType& particles, const double dt )
+    {
+        _euler_timer.start();
+        auto model = _model;
+        const auto rho = particles.sliceDensity();
+        const auto conduction = particles.sliceTemperatureConduction();
+        auto temp = particles.sliceTemperature();
+        auto n_local = particles.n_local;
+        auto euler_func = KOKKOS_LAMBDA( const int i )
+        {
+            temp( i ) += dt / rho( i ) / model.cp * conduction( i );
```

Comment:
Do we have somewhere a heat source term (like a body force)? We do not need it for the current example, but we would need it for b.c..
I am referring to the term w_i^n in the notes (see below):
![fig](https://github.com/user-attachments/assets/53c2b25c-accc-45a1-8114-28582a6c4504)

Reply:
That would just be a non-force body term